### ISSUE TITLE
[BUG] float division by zero (swimming activities)

### ISSUE THREAD

**Issue by flare04** [CTHRU/Hitrava#8]
**Describe the bug**
Whilst converting my backed up data, some events cause the program to crash. All 4/5 files that have the issue have a sportstype of 102

**Logs**
2020-05-01 22:58:57,327 - DEBUG - _get_pool_swim_data - Calculated swim data for lap 30 : {'lap': 30, 'swolf': 39, 'strokes': 12, 'duration': 27, 'distance': 25.0, 'speed': 0.9259259259259259, 'start': datetime.datetime(2019, 12, 4, 1, 14), 'stop': datetime.datetime(2019, 12, 4, 1, 14, 27)}
2020-05-01 22:58:57,327 - ERROR - generate_xml - Error generating TCX XML content for activity <HiTrack_20191204_120058>
float division by zero
Traceback (most recent call last):
  File "./Hitrava.py", line 1322, in generate_xml
    self._generate_swim_xml_data(el_activity)
  File "./Hitrava.py", line 1437, in _generate_swim_xml_data
    for n, lap in enumerate(self.hi_activity.get_swim_data()):
  File "./Hitrava.py", line 705, in get_swim_data
    return self._get_pool_swim_data()
  File "./Hitrava.py", line 765, in _get_pool_swim_data
    lap_data['speed'] = self.pool_length / lap_data['duration']
ZeroDivisionError: float division by zero

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./Hitrava.py", line 1800, in <module>
    main()
  File "./Hitrava.py", line 1795, in main
    tcx_activity.save()
  File "./Hitrava.py", line 1500, in save
    self.generate_xml()
  File "./Hitrava.py", line 1367, in generate_xml
    raise Exception('Error generating TCX XML content for activity <%s>\n%s', self.hi_activity.activity_id, e)
Exception: ('Error generating TCX XML content for activity <%s>\n%s', 'HiTrack_20191204_120058', ZeroDivisionError('float division by zero'))
2020-05-01 22:58:57,347 - DEBUG - _close_json - JSON file <./output/Motion path detail data & description/motion path detail data.json> closed

**Files**

**Comment by CTHRU** [comment]
@flare04 - Thank you for reporting the issue. I have a temporary copy of your json file. If you want, you can remove it from your post. Could you do me a favour and check in Huawei Health/let me know whether your activity of December 4th, 2019 is actually an indoor swimming activity? This will help me to allow to set the sport type correctly from the JSON data. I will look into your issue at a later time and get back to you soon.

**Comment by flare04** [comment]
Hi yes it was an indoor swim. 25m pool, 44 laps.

**Comment by CTHRU** [comment]
Just a heads-up, I looked at your file and I noticed that probably Huawei made some adjustments to the way swimming data is stored in the file, which probably means the current way of detecting swimming laps can't be used anymore in a reliable fashion. 

The good news is that the required data is at another place in the file, but it will take me some time to create new code to parse it and integrate it in the current code. I will work on it, but can't promise you a release date for now.
